Add option to close related issues on manual grading

jonatanschroeder opened the pull request:
Closes #9678. Given many cases of manual grading involve resolving an issue raised by a student, this change allows the issue to be closed as the manual grading is submitted.

I chose to close the issue outside of the manual grading transaction, though I'm open to moving the logic to the manual grading process if that makes sense.

![image](https://github.com/PrairieLearn/PrairieLearn/assets/1004907/ae800fe8-551b-4344-8b17-596585680029)

nwalters512 reviewed (approved):
Not requesting a change in this PR, but a UX note: after issues are closed, one can revisit the manual grading page and still see the issues there, with no indication that thew were closed, and it's not immediately obvious why the "close issue" checkboxes don't show up again. It'd be nice to consider collapsing closed issues a la resolved comments on GitHub.

Very tangential comment: we should add `closed_at`/`closed_by` to `issues` so that we can display that to other instructors.

jonatanschroeder commented:
> Not requesting a change in this PR, but a UX note: after issues are closed, one can revisit the manual grading page and still see the issues there, with no indication that thew were closed, and it's not immediately obvious why the "close issue" checkboxes don't show up again. It'd be nice to consider collapsing closed issues a la resolved comments on GitHub.

Yup, I'm aware of this. Somewhat related to #3142. I added a note there.

> Very tangential comment: we should add `closed_at`/`closed_by` to `issues` so that we can display that to other instructors.

Added to #4093.
